test/fuzz/test_runner.py

```diff
@@ -213,7 +211,27 @@ def generate_corpus(*, fuzz_pool, src_dir, build_dir, corpus_dir, targets):
             text=True,
         ).stdout.splitlines()
         lines = [l.split("\"", 1)[1].split("\"")[0] for l in lines if l.startswith("src/test/fuzz/rpc.cpp-    \"")]
+        assert len(lines)
         targets += [(rpc_target, {"LIMIT_TO_RPC_COMMAND": r}) for r in lines]
+        targets += [(rpc_target, {})]  # Allow for cross-pollination, or unlimited search
+    if has_p2p_msg:
+        with open(os.path.join(src_dir, "src", "protocol.cpp"), "r", encoding="utf8") as f:
```

Comment:
Could make this a bit more robust by only considering lines within `namespace NetMsgType`?

```py
    if has_p2p_msg:
        cmd = ["awk" , "/namespace NetMsgType {/,/} \/\/ namespace NetMsgType/ { print $0 }", str(Path(src_dir) / 'src' / 'protocol.cpp')]
        lines = subprocess.run(
            cmd,
            check=True,
            stdout=subprocess.PIPE,
            text=True,
        ).stdout.splitlines()
        lines = [l.split('"', 1)[1].split('"')[0] for l in lines if l.startswith("const char")]
```

Reply:
Works fine here (macOS 13.4) but I'm not familiar with its portability, so if that's been an issue in the past then yeah might not be ideal. I agree that being too greedy is not a huge risk but if we can be specific I think that's still preferred. Alternatively, could also use `git grep` with `g_all_net_message_types` like we do for the `rpc` target - didn't see that earlier.